metricbeat/tests/system/test_beat.py

```diff
@@ -39,9 +40,13 @@ def test_xpack(self):
         # it's cluster UUID in the process. Otherwise, the monitoring Metricbeat instance will
         # show errors in its log about not being able to determine the Elasticsearch cluster UUID
         # to be associated with the monitored Metricbeat instance.
-        time.sleep(30)
+        self.wait_until(cond=self.mb_index_exists, max_timeout=60)
 
         proc = self.start_beat()
         self.wait_until(lambda: self.output_lines() > 0)
         proc.check_kill_and_wait()
         self.assert_no_logged_warnings()
+
+    def mb_index_exists(self):
+        es = Elasticsearch([self.get_elasticsearch_url()])
+        return len(es.cat.indices(index='metricbeat-*', h='index')) > 0
```

Comment:
Hey @jsoriano, I could use some guidance here. CI is failing on this line with this error:

```
17:00:38 ======================================================================
17:00:38 ERROR: beat-xpack module tests
17:00:38 ----------------------------------------------------------------------
17:00:38 Traceback (most recent call last):
17:00:38   File "/go/src/github.com/elastic/beats/metricbeat/tests/system/test_beat.py", line 43, in test_xpack
17:00:38     self.wait_until(cond=self.mb_index_exists, max_timeout=60)
17:00:38   File "/go/src/github.com/elastic/beats/libbeat/tests/system/beat/beat.py", line 349, in wait_until
17:00:38     while not cond():
17:00:38   File "/go/src/github.com/elastic/beats/metricbeat/tests/system/test_beat.py", line 52, in mb_index_exists
17:00:38     return len(es.cat.indices(index='metricbeat-*', h='index')) > 0
17:00:38   File "/go/src/github.com/elastic/beats/metricbeat/build/python-env/local/lib/python2.7/site-packages/elasticsearch/client/utils.py", line 76, in _wrapped
17:00:38     return func(*args, params=params, **kwargs)
17:00:38   File "/go/src/github.com/elastic/beats/metricbeat/build/python-env/local/lib/python2.7/site-packages/elasticsearch/client/cat.py", line 137, in indices
17:00:38     'indices', index), params=params)
17:00:38   File "/go/src/github.com/elastic/beats/metricbeat/build/python-env/local/lib/python2.7/site-packages/elasticsearch/transport.py", line 318, in perform_request
17:00:38     status, headers_response, data = connection.perform_request(method, url, params, body, headers=headers, ignore=ignore, timeout=timeout)
17:00:38   File "/go/src/github.com/elastic/beats/metricbeat/build/python-env/local/lib/python2.7/site-packages/elasticsearch/connection/http_urllib3.py", line 181, in perform_request
17:00:38     raise ConnectionError('N/A', str(e), e)
17:00:38 ConnectionError: ConnectionError(<urllib3.connection.HTTPConnection object at 0x7f23baa492d0>: Failed to establish a new connection: [Errno 111] Connection refused) caused by: NewConnectionError(<urllib3.connection.HTTPConnection object at 0x7f23baa492d0>: Failed to establish a new connection: [Errno 111] Connection refused)
...
17:00:38 NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f23baa492d0>: Failed to establish a new connection: [Errno 111] Connection refused
17:00:38 elasticsearch: DEBUG: > None
17:00:38 elasticsearch: WARNING: GET http://localhost:9200/_cat/indices/metricbeat-*?h=index [status:N/A request:0.001s]
```

I believe the problem is that `self.get_elasticsearch_url()` is returning `http://localhost:9200`, which is not how to address the Elasticsearch service running in the Docker container. Do you know the right way to address it?

Reply:
Never mind, I figured it out:

```python
self.compose_host("elasticsearch")
```